[Excel] (Custom Functions) Update XLL article

alison-mk opened the pull request:
Addressing issue #2837.

opbld31 commented:
Docs Build status updates of commit _[e02d776](https://github.com/OfficeDev/office-js-docs-pr/commits/e02d7766134e01a18c9c4fbeeaa4a8db15f133c9)_: 

### :white_check_mark: Validation status: passed


File | Status | Preview URL | Details
---- | ------ | ----------- | -------
[docs/excel/make-custom-functions-compatible-with-xll-udf.md](https://github.com/OfficeDev/office-js-docs-pr/blob/alison-mk-cf-xll/docs/excel/make-custom-functions-compatible-with-xll-udf.md) | :white_check_mark:Succeeded | [View](https://review.docs.microsoft.com/en-us/office/dev/add-ins/excel/make-custom-functions-compatible-with-xll-udf?branch=pr-en-us-2895) |

For more details, please refer to the [build report](https://opbuilduserstoragepublic.blob.core.windows.net/report/2021%5C8%5C24%5C57856ef7-e8cf-54ca-0d12-df5f42d1e8bb%5CPullRequest%5C202108240006257943-2895%5Cworkflow_report.html?sv=2018-03-28&sr=b&sig=lU4ZxpUSYHkyVtmhX7LQqX%2FmlRK%2F%2BjyTD42zK%2BnNDj8%3D&st=2021-08-24T00%3A02%3A05Z&se=2021-09-24T00%3A07%3A05Z&sp=r).

**Note:** Broken links written as relative paths are included in the above build report. For broken links written as absolute paths or external URLs, see the [broken link report](https://ops.microsoft.com/#/repos/57856ef7-e8cf-54ca-0d12-df5f42d1e8bb?tabName=brokenlinks).

For any questions, please:<ul><li>Try searching the docs.microsoft.com <a href="https://review.docs.microsoft.com/en-us/help/?branch=master">contributor guides</a></li><li>Post your question in the <a href="https://teams.microsoft.com/l/channel/19%3a7ecffca1166a4a3986fed528cf0870ee%40thread.skype/General?groupId=de9ddba4-2574-4830-87ed-41668c07a1ca&amp;tenantId=72f98bf-86f1-41af-91ab-2d7cd011db47">Docs support channel</a></li></ul>

opbld31 commented:
Docs Build status updates of commit _[1800b06](https://github.com/OfficeDev/office-js-docs-pr/commits/1800b061075c613eae54f5cf95b7c80012d06b9b)_: 

### :white_check_mark: Validation status: passed


File | Status | Preview URL | Details
---- | ------ | ----------- | -------
[docs/excel/make-custom-functions-compatible-with-xll-udf.md](https://github.com/OfficeDev/office-js-docs-pr/blob/alison-mk-cf-xll/docs/excel/make-custom-functions-compatible-with-xll-udf.md) | :white_check_mark:Succeeded | [View](https://review.docs.microsoft.com/en-us/office/dev/add-ins/excel/make-custom-functions-compatible-with-xll-udf?branch=pr-en-us-2895) |

For more details, please refer to the [build report](https://opbuilduserstoragepublic.blob.core.windows.net/report/2021%5C8%5C24%5C57856ef7-e8cf-54ca-0d12-df5f42d1e8bb%5CPullRequest%5C202108240029257864-2895%5Cworkflow_report.html?sv=2018-03-28&sr=b&sig=l1HADKzwb2JOYWNUWKfokBFEddUMudJDmKGCdF17%2FIM%3D&st=2021-08-24T00%3A25%3A11Z&se=2021-09-24T00%3A30%3A11Z&sp=r).

**Note:** Broken links written as relative paths are included in the above build report. For broken links written as absolute paths or external URLs, see the [broken link report](https://ops.microsoft.com/#/repos/57856ef7-e8cf-54ca-0d12-df5f42d1e8bb?tabName=brokenlinks).

For any questions, please:<ul><li>Try searching the docs.microsoft.com <a href="https://review.docs.microsoft.com/en-us/help/?branch=master">contributor guides</a></li><li>Post your question in the <a href="https://teams.microsoft.com/l/channel/19%3a7ecffca1166a4a3986fed528cf0870ee%40thread.skype/General?groupId=de9ddba4-2574-4830-87ed-41668c07a1ca&amp;tenantId=72f98bf-86f1-41af-91ab-2d7cd011db47">Docs support channel</a></li></ul>

alison-mk commented:
@Rick-Kirkham I'm working on addressing the issues you noted in #2837. Can you take a look at this PR? Open to feedback from you, or feel free to contribute commits directly to this PR.

In particular, I'm not sure how to distinguish XLL user-defined functions, XLL compatible custom functions, and Excel add-in custom functions. 

Thanks!

Rick-Kirkham commented on docs/excel/make-custom-functions-compatible-with-xll-udf.md:
```diff
@@ -1,30 +1,27 @@
-> - Excel on Mac (version 13.329 or later)
->
-> To use COM add-in and XLL UDF compatibility within Excel on the web, login by using either your Microsoft 365 subscription or a [Microsoft account](https://account.microsoft.com/account). If you don't already have a Microsoft 365 subscription, you can a free, 90-day renewable Microsoft 365 subscription by joining the [Microsoft 365 developer program](https://developer.microsoft.com/office/dev-p ...
+> An XLL add-in is an Excel add-in file with the file extension **.xll**. An XLL file is a type of dynamic link library (DLL) file that can only be opened by Excel. XLL add-in files must be written in C or C++. 
```
Can we get this definition earlier? Maybe it could be a note preceding the top paragraph, so readers know what this is before we start talking about it. A link to dev docs for XLL would be nice too.

Rick-Kirkham commented on docs/excel/make-custom-functions-compatible-with-xll-udf.md:
```diff
@@ -44,15 +41,15 @@ The following example shows how to specify both a COM add-in and an XLL as equiv
 
 > [!NOTE]
-> If an add-in declares its custom functions to be XLL compatible, changing the manifest at a later time could break a user's workbook because it will change the file format.
+> If an Excel JavaScript API add-in declares its custom functions to be compatible with an XLL add-in, changing the manifest at a later time could break a user's workbook because it will change the file format.
```
This paragraph could use some elaboration. Exactly what changes in the manifest would have this effect? Does this mean that once an add-in declares its custom functions to be compatible with an XLL, it can never reverse this declaration? In what sense of format is the workbook's file format changed?

Rick-Kirkham commented on docs/excel/make-custom-functions-compatible-with-xll-udf.md:
```diff
@@ -44,15 +41,15 @@ The following example shows how to specify both a COM add-in and an XLL as equiv
 ## Custom function behavior for XLL compatible functions
 
-An add-in's XLL functions are converted to XLL compatible custom functions when a spreadsheet is opened and there is an equivalent add-in available. On the next save, the XLL functions are written to the file in a compatible mode so that they work with both the XLL and Excel add-in custom functions (when on other platforms).
+An add-in's XLL functions are converted to XLL compatible custom functions when a spreadsheet is opened and there is an equivalent add-in available. On the next save, the XLL functions are written to the file in a compatible mode so that they work with both the XLL add-in and Excel JavaScript API add-in custom functions (when on other platforms).
```
What is this conversion all about? Are the XLL custom functions converted to JavaScript custom functions?

Rick-Kirkham commented:
> @Rick-Kirkham I'm working on addressing the issues you noted in #2837. Can you take a look at this PR? Open to feedback from you, or feel free to contribute commits directly to this PR.
> 
> In particular, I'm not sure how to distinguish XLL user-defined functions, XLL compatible custom functions, and Excel add-in custom functions.
> 
> Thanks!

I made some comments. I also don't know how to distinguish the 3 because I don't understand what the "XLL compatible custom functions" are. You may need to track down the original subject matter expert who made this distinction.

opbld33 commented:
Docs Build status updates of commit _[5663ffd](https://github.com/OfficeDev/office-js-docs-pr/commits/5663ffdf2555a76871771336cb7d81db2df3da38)_: 

### :white_check_mark: Validation status: passed


File | Status | Preview URL | Details
---- | ------ | ----------- | -------
[docs/excel/make-custom-functions-compatible-with-xll-udf.md](https://github.com/OfficeDev/office-js-docs-pr/blob/alison-mk-cf-xll/docs/excel/make-custom-functions-compatible-with-xll-udf.md) | :white_check_mark:Succeeded | [View](https://review.docs.microsoft.com/en-us/office/dev/add-ins/excel/make-custom-functions-compatible-with-xll-udf?branch=pr-en-us-2895) |

For more details, please refer to the [build report](https://opbuilduserstoragepublic.blob.core.windows.net/report/2021%5C8%5C24%5C57856ef7-e8cf-54ca-0d12-df5f42d1e8bb%5CPullRequest%5C202108242152121495-2895%5Cworkflow_report.html?sv=2018-03-28&sr=b&sig=21DX1gK010OnBpa%2Bbjb4KKmHvNLifHJEazpqfPFg2BM%3D&st=2021-08-24T21%3A47%3A51Z&se=2021-09-24T21%3A52%3A51Z&sp=r).

**Note:** Broken links written as relative paths are included in the above build report. For broken links written as absolute paths or external URLs, see the [broken link report](https://ops.microsoft.com/#/repos/57856ef7-e8cf-54ca-0d12-df5f42d1e8bb?tabName=brokenlinks).

For any questions, please:<ul><li>Try searching the docs.microsoft.com <a href="https://review.docs.microsoft.com/en-us/help/?branch=master">contributor guides</a></li><li>Post your question in the <a href="https://teams.microsoft.com/l/channel/19%3a7ecffca1166a4a3986fed528cf0870ee%40thread.skype/General?groupId=de9ddba4-2574-4830-87ed-41668c07a1ca&amp;tenantId=72f98bf-86f1-41af-91ab-2d7cd011db47">Docs support channel</a></li></ul>

opbld32 commented:
Docs Build status updates of commit _[00be84d](https://github.com/OfficeDev/office-js-docs-pr/commits/00be84d5963e50d9cff72fe4bd468a26b64f94fe)_: 

### :white_check_mark: Validation status: passed


File | Status | Preview URL | Details
---- | ------ | ----------- | -------
[docs/excel/make-custom-functions-compatible-with-xll-udf.md](https://github.com/OfficeDev/office-js-docs-pr/blob/alison-mk-cf-xll/docs/excel/make-custom-functions-compatible-with-xll-udf.md) | :white_check_mark:Succeeded | [View](https://review.docs.microsoft.com/en-us/office/dev/add-ins/excel/make-custom-functions-compatible-with-xll-udf?branch=pr-en-us-2895) |

For more details, please refer to the [build report](https://opbuilduserstoragepublic.blob.core.windows.net/report/2021%5C8%5C24%5C57856ef7-e8cf-54ca-0d12-df5f42d1e8bb%5CPullRequest%5C202108242200003961-2895%5Cworkflow_report.html?sv=2018-03-28&sr=b&sig=nPcecCw4kZkWWX%2Ff52PX7CAMnwsUhwv%2FsfNBKUwe3sc%3D&st=2021-08-24T21%3A55%3A48Z&se=2021-09-24T22%3A00%3A48Z&sp=r).

**Note:** Broken links written as relative paths are included in the above build report. For broken links written as absolute paths or external URLs, see the [broken link report](https://ops.microsoft.com/#/repos/57856ef7-e8cf-54ca-0d12-df5f42d1e8bb?tabName=brokenlinks).

For any questions, please:<ul><li>Try searching the docs.microsoft.com <a href="https://review.docs.microsoft.com/en-us/help/?branch=master">contributor guides</a></li><li>Post your question in the <a href="https://teams.microsoft.com/l/channel/19%3a7ecffca1166a4a3986fed528cf0870ee%40thread.skype/General?groupId=de9ddba4-2574-4830-87ed-41668c07a1ca&amp;tenantId=72f98bf-86f1-41af-91ab-2d7cd011db47">Docs support channel</a></li></ul>

alison-mk commented:
@Rick-Kirkham Thanks for the review! I will consult @davidchesnut upon his return. I've recorded your remaining concerns in an [internal work item](https://office.visualstudio.com/OC/_workitems/edit/5368630). I'd like to merge this PR in the interim, because I think the new !NOTE and term clarification is helpful. Do you agree?

Rick-Kirkham commented:
> @Rick-Kirkham Thanks for the review! I will consult @davidchesnut upon his return. I've recorded your remaining concerns in an [internal work item](https://office.visualstudio.com/OC/_workitems/edit/5368630). I'd like to merge this PR in the interim, because I think the new !NOTE and term clarification is helpful. Do you agree?

OK. Go ahead and merge.
